website/project/model.py

```diff
@@ -2620,3 +2769,226 @@ def to_json(self):
+                    'node': parent_registration._id,
+                    'retraction_id': self._id,
+                },
+                auth=Auth(user),
+                log_date=datetime.datetime.utcnow(),
+                save=True,
+            )
+            # Remove any embargoes associated with the registration
+            if parent_registration.embargo_end_date or parent_registration.pending_embargo:
+                parent_registration.embargo.state = self.CANCELLED
+                parent_registration.add_log(
+                    action=NodeLog.EMBARGO_CANCELLED,
+                    params={
+                        'node': parent_registration._id,
+                        'embargo_id': parent_registration.embargo._id,
+                    },
+                    auth=Auth(user),
+                    log_date=datetime.datetime.utcnow(),
+                    save=False,
+                )
+                parent_registration.embargo.save()
+            # Ensure retracted registration is public
+            if not parent_registration.is_public:
+                parent_registration.set_privacy('public')
+
+
+def validate_embargo_state(value):
+    acceptable_states = [
+        Embargo.UNAPPROVED, Embargo.ACTIVE, Embargo.CANCELLED, Embargo.COMPLETED
+    ]
+    if value not in acceptable_states:
+        raise ValidationValueError('Invalid embargo state assignment.')
+    return True
+
+
+class Embargo(StoredObject):
+    """Embargo object for registrations waiting to go public."""
+
+    UNAPPROVED = 'unapproved'
+    ACTIVE = 'active'
+    CANCELLED = 'cancelled'
+    COMPLETED = 'completed'
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+    initiation_date = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow)
+    initiated_by = fields.ForeignField('user', backref='embargoed')
+    end_date = fields.DateTimeField()
+    # Expanded: Dictionary field mapping admin IDs their approval status and relevant tokens:
+    # {
+    #   'b3k97': {
+    #     'has_approved': False,
+    #     'approval_token': 'Pew7wj1Puf7DENUPFPnXSwa1rf3xPN',
+    #     'disapproval_token': 'TwozClTFOic2PYxHDStby94bCQMwJy'}
+    # }
+    approval_state = fields.DictionaryField()
+    # One of 'unapproved', 'active', 'cancelled', or 'completed
+    state = fields.StringField(default='unapproved', validate=validate_embargo_state)
+    for_existing_registration = fields.BooleanField(default=False)
+
+    def __repr__(self):
+        parent_registration = Node.find_one(Q('embargo', 'eq', self))
+        return ('<Embargo(parent_registration={0}, initiated_by={1}, '
+                'end_date={2}) with _id {3}>').format(
+            parent_registration,
+            self.initiated_by,
+            self.end_date,
+            self._id
+        )
+
+    @property
+    def embargo_end_date(self):
+        if self.state == 'active':
+            return self.end_date.strftime("%A, %b. %d, %Y")  # e.g. 'Friday, Jan. 01, 2016'
```

Comment:
Originally this was `is_embargoed` but we had a need for having an embargo's end date in the templates. Obviously Datetime objects can't be serialized so I converted end_date to return the string representation of the embargoes end date to avoid adding **another** thing to the serializer as the fact that an embargo has an end date implies that is embargoed. So I see two options moving forward: Either make this a really long, informative name that can be used for both use cases e.g. `formatted_embargo_end_date` **or** break it into two fields `is_embargoed` and `formatted_embargo_end_date` as this will appear as Node.formatted_embargo_end_date rather that Node.embargo.formatted_end_date, modify the serializer and all other bits of the code accordingly.  Thoughts?